File: apps/android/app/src/main/java/com/app/rasifiters/ui/summary/SummaryCards.kt
package com.app.rasifiters.ui.summary

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowDownward
import androidx.compose.material.icons.filled.ArrowUpward
import androidx.compose.material.icons.filled.Bedtime
import androidx.compose.material.icons.filled.Bolt
import androidx.compose.material.icons.filled.ChevronRight
import androidx.compose.material.icons.filled.Add
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.app.rasifiters.core.theme.AppBlue
import com.app.rasifiters.core.theme.AppBlueLight
import com.app.rasifiters.core.theme.AppGreen
import com.app.rasifiters.core.theme.AppOrange
import com.app.rasifiters.core.theme.AppOrangeGradientEnd
import com.app.rasifiters.core.theme.AppRed
import java.util.Locale
import kotlin.math.abs
import kotlin.math.roundToInt

// Faithful 1:1 port of the iOS Summary cards (Features/Home/Tabs/SummaryCards.swift), rendered in the
// Material idiom: a flat rounded-surface card in place of iOS's glassy CardShell (blur has no cheap
// Compose analog and reads as noise on Android). Same content, same numbers, same change badges.

/** The shared rounded-surface card container — the Material analog of iOS `CardShell`. */
@Composable
fun SummaryCard(
    modifier: Modifier = Modifier,
    height: androidx.compose.ui.unit.Dp? = null,
    content: @Composable () -> Unit,
) {
    var m = modifier
        .fillMaxWidth()
        .clip(RoundedCornerShape(20.dp))
        .background(MaterialTheme.colorScheme.surface)
        .border(1.dp, MaterialTheme.colorScheme.onSurface.copy(alpha = 0.08f), RoundedCornerShape(20.dp))
    if (height != null) m = m.height(height)
    Box(modifier = m.padding(16.dp)) { content() }
}

// MARK: - Program progress

@Composable
fun ProgramProgressCard(progress: Int, elapsedDays: Int, totalDays: Int, status: String) {
    SummaryCard(height = 240.dp) {
        Column(modifier = Modifier.fillMaxWidth()) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text(
                    "Program Progress",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.SemiBold,
                    modifier = Modifier.weight(1f),
                )
                Box(
                    modifier = Modifier
                        .clip(CircleShape)
                        .background(AppOrange.copy(alpha = 0.18f))
                        .padding(horizontal = 10.dp, vertical = 4.dp),
                ) {
                    Text(
                        status.uppercase(Locale.US),
                        style = MaterialTheme.typography.labelMedium,
                        fontWeight = FontWeight.Bold,
                        color = AppOrange,
                    )
                }
            }
            Box(modifier = Modifier.fillMaxWidth().weight(1f), contentAlignment = Alignment.Center) {
                CompletionRing(progress = progress, modifier = Modifier.size(140.dp))
                Column(horizontalAlignment = Alignment.CenterHorizontally) {
                    Text("$progress%", style = MaterialTheme.typography.headlineMedium, fontWeight = FontWeight.Bold)
                    Text(
                        "$elapsedDays/$totalDays days",
                        style = MaterialTheme.typography.bodyMedium,
                        fontWeight = FontWeight.SemiBold,
                        color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f),
                    )
                }
            }
        }
    }
}

@Composable
private fun CompletionRing(progress: Int, modifier: Modifier = Modifier) {
    val track = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.12f)
    val fraction = (progress.coerceIn(0, 100)) / 100f
    Canvas(modifier = modifier) {
        val stroke = 12.dp.toPx()
        val inset = stroke / 2
        val arcSize = Size(size.width - stroke, size.height - stroke)
        drawArc(
            color = track,
            startAngle = 0f,
            sweepAngle = 360f,
            useCenter = false,
            topLeft = Offset(inset, inset),
            size = arcSize,
            style = Stroke(width = stroke),
        )
        drawArc(
            brush = Brush.linearGradient(listOf(AppOrange, AppOrangeGradientEnd)),
            startAngle = -90f,
            sweepAngle = 360f * fraction,
            useCenter = false,
            topLeft = Offset(inset, inset),
            size = arcSize,
            style = Stroke(width = stroke, cap = StrokeCap.Round),
        )
    }
}

// MARK: - Change badge (shared by the 4 MTD metric cards; iOS duplicates it inline per card, F4)

@Composable
private fun ChangeBadge(change: Double) {
    val up = change >= 0
    val color = if (up) AppGreen else AppRed
    Row(
        modifier = Modifier
            .clip(RoundedCornerShape(10.dp))
            .background(color.copy(alpha = 0.12f))
            .padding(horizontal = 10.dp, vertical = 6.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(4.dp),
    ) {
        Icon(
            if (up) Icons.Filled.ArrowUpward else Icons.Filled.ArrowDownward,
            contentDescription = null,
            tint = color,
            modifier = Modifier.size(14.dp),
        )
        Text(
            String.format(Locale.US, "%.1f%%", abs(change)),
            style = MaterialTheme.typography.labelLarge,
            fontWeight = FontWeight.SemiBold,
            color = color,
        )
    }
}

/** A tall metric card: title · big value · change badge · "vs prior MTD". Optional secondary line (MTD). */
@Composable
private fun MetricCard(title: String, value: String, secondary: String?, change: Double) {
    SummaryCard(height = 240.dp) {
        Column(modifier = Modifier.fillMaxWidth()) {
            Text(title, style = MaterialTheme.typography.titleSmall, fontWeight = FontWeight.SemiBold)
            Spacer(Modifier.weight(1f))
            Text(value, style = MaterialTheme.typography.headlineSmall, fontWeight = FontWeight.Bold)
            if (secondary != null) {
                Spacer(Modifier.height(4.dp))
                Text(
                    secondary,
                    style = MaterialTheme.typography.bodySmall,
                    fontWeight = FontWeight.SemiBold,
                    color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f),
                )
            }
            Spacer(Modifier.weight(1f))
            ChangeBadge(change)
            Spacer(Modifier.height(6.dp))
            Text(
                "vs prior MTD",
                style = MaterialTheme.typography.bodySmall,
                color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f),
            )
        }
    }
}

@Composable
fun MtdParticipationCard(active: Int, total: Int, pct: Double, change: Double, modifier: Modifier = Modifier) {
    Box(modifier) {
        MetricCard("MTD Participation", "${pct.roundToInt()}%", "$active/$total members active", change)
    }
}

@Composable
fun TotalWorkoutsCard(total: Int, change: Double, modifier: Modifier = Modifier) {
    Box(modifier) { MetricCard("Total workouts", "$total", null, change) }
}

@Composable
fun TotalDurationCard(hours: Double, change: Double, modifier: Modifier = Modifier) {
    val whole = hours.roundToInt()
    val label = if (abs(hours - whole) < 0.05) "$whole" else String.format(Locale.US, "%.1f", hours)
    Box(modifier) { MetricCard("Total duration", "$label hrs", null, change) }
}

@Composable
fun AvgDurationCard(minutes: Int, change: Double, modifier: Modifier = Modifier) {
    Box(modifier) { MetricCard("Avg duration", "$minutes mins", null, change) }
}

/** A tall placeholder while a metric card's data is still loading (iOS `PlaceholderCard`). */
@Composable
fun MetricPlaceholderCard(title: String, modifier: Modifier = Modifier) {
    Box(modifier) {
        SummaryCard(height = 240.dp) {
            Column(
                modifier = Modifier.fillMaxWidth(),
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.Center,
            ) {
                Text(
                    title,
                    style = MaterialTheme.typography.titleSmall,
                    fontWeight = FontWeight.SemiBold,
                    color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f),
                )
            }
        }
    }
}

// MARK: - Action cards (gradient; navigate to the log forms unless data-entry is locked)

@Composable
fun AddWorkoutCard(locked: Boolean, onClick: () -> Unit) {
    ActionCard(
        gradient = Brush.linearGradient(listOf(AppOrange, AppOrangeGradientEnd)),
        icon = Icons.Filled.Add,
        iconTint = Color.Black,
        title = "Add workouts",
        subtitle = "Log one or many sessions at once and keep progress up to date.",
        pillLabel = "Log sessions",
        pillIcon = Icons.Filled.Bolt,
        pillBg = AppOrange,
        onBrand = Color.Black,
        locked = locked,
        onClick = onClick,
    )
}

@Composable
fun AddDailyHealthCard(locked: Boolean, onClick: () -> Unit) {
    ActionCard(
        gradient = Brush.linearGradient(listOf(AppBlue, AppBlueLight)),
        icon = Icons.Filled.Bedtime,
        iconTint = Color.White,
        title = "Log daily health",
        subtitle = "Track sleep hours and diet quality for the day.",
        pillLabel = "Log day",
        pillIcon = Icons.Filled.Add,
        pillBg = Color.White.copy(alpha = 0.2f),
        onBrand = Color.White,
        locked = locked,
        onClick = onClick,
    )
}

@Composable
private fun ActionCard(
    gradient: Brush,
    icon: ImageVector,
    iconTint: Color,
    title: String,
    subtitle: String,
    pillLabel: String,
    pillIcon: ImageVector,
    pillBg: Color,
    onBrand: Color,
    locked: Boolean,
    onClick: () -> Unit,
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .height(200.dp)
            .alpha(if (locked) 0.5f else 1f)
            .clip(RoundedCornerShape(20.dp))
            .background(gradient)
            .clickable(enabled = !locked, onClick = onClick)
            .padding(16.dp),
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Box(
                modifier = Modifier.size(36.dp).clip(CircleShape).background(Color.White.copy(alpha = 0.2f)),
                contentAlignment = Alignment.Center,
            ) {
                Icon(icon, contentDescription = null, tint = iconTint, modifier = Modifier.size(18.dp))
            }
            Spacer(Modifier.weight(1f))
            Icon(Icons.Filled.ChevronRight, contentDescription = null, tint = onBrand.copy(alpha = 0.7f))
        }
        Spacer(Modifier.height(10.dp))
        Text(title, style = MaterialTheme.typography.titleLarge, fontWeight = FontWeight.Bold, color = onBrand)
        Spacer(Modifier.height(4.dp))
        Text(subtitle, style = MaterialTheme.typography.bodyMedium, color = onBrand.copy(alpha = 0.7f))
        Spacer(Modifier.weight(1f))
        Row(
            modifier = Modifier
                .clip(CircleShape)
                .background(pillBg)
                .padding(horizontal = 16.dp, vertical = 8.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(6.dp),
        ) {
            Icon(pillIcon, contentDescription = null, tint = onBrand, modifier = Modifier.size(16.dp))
            Text(pillLabel, style = MaterialTheme.typography.labelLarge, fontWeight = FontWeight.SemiBold, color = onBrand)
        }
    }
}

File: apps/ios/RaSi-Fiters-App/Features/Home/Tabs/SummaryCards.swift
import SwiftUI

// MARK: - Shared building blocks
// Faithful 1:1 port of the legacy iOS Summary cards (ios-mobile Features/Home/Helpers/AdminHomeHelpers.swift).

struct SummaryHeader: View {
    let title: String
    let subtitle: String
    let status: String
    let initials: String

    var body: some View {
        HStack(alignment: .center, spacing: 14) {
            VStack(alignment: .leading, spacing: 6) {
                Text(title)
                    .font(.largeTitle.weight(.bold))
                    .foregroundColor(Color(.label))
                Text("\(subtitle)")
                    .font(.headline.weight(.semibold))
                    .foregroundColor(Color(.secondaryLabel))
            }

            Spacer()

            ZStack {
                Circle()
                    .fill(
                        LinearGradient(
                            colors: [Color.appOrange, Color.appOrangeGradientEnd],
                            startPoint: .topLeading,
                            endPoint: .bottomTrailing
                        )
                    )
                    .frame(width: 52, height: 52)
                Text(initials)
                    .font(.headline.weight(.bold))
                    .foregroundColor(.black)
            }
        }
    }
}

struct CardShell<Content: View, Background: View>: View {
    let background: Background
    var strokeColor: Color = Color(.white).opacity(0.35)
    var height: CGFloat = 240
    @ViewBuilder var content: () -> Content

    var body: some View {
        content()
            .padding(16)
            .frame(maxWidth: .infinity, alignment: .topLeading)
            .frame(height: height, alignment: .topLeading)
            .background(
                ZStack {
                    background
                        .blur(radius: 12)
                    RoundedRectangle(cornerRadius: 20, style: .continuous)
                        .fill(Color(.white).opacity(0.25))
                        .background(
                            RoundedRectangle(cornerRadius: 20, style: .continuous)
                                .fill(Color(.systemBackground).opacity(0.3))
                        )
                }
                .clipShape(RoundedRectangle(cornerRadius: 20, style: .continuous))
            )
            .overlay(
                RoundedRectangle(cornerRadius: 20, style: .continuous)
                    .stroke(strokeColor, lineWidth: 1)
            )
            .shadow(color: Color(.black).opacity(0.06), radius: 10, x: 0, y: 6)
    }
}

struct PlaceholderCard: View {
    let title: String

    var body: some View {
        VStack(alignment: .center, spacing: 6) {
            Text(title)
                .font(.subheadline.weight(.semibold))
                .foregroundColor(Color(.secondaryLabel))
            ProgressView()
        }
        .frame(maxWidth: .infinity)
        .padding()
        .background(
            RoundedRectangle(cornerRadius: 18, style: .continuous)
                .fill(Color(.systemBackground).opacity(0.9))
        )
        .overlay(
            RoundedRectangle(cornerRadius: 18, style: .continuous)
                .stroke(Color(.systemGray4).opacity(0.6), lineWidth: 1)
        )
    }
}

// MARK: - Program progress

struct ProgramProgressCard: View {
    let progress: Int
    let elapsedDays: Int
    let totalDays: Int
    let remainingDays: Int
    let status: String

    var body: some View {
        CardShell(
            background: Color(.systemBackground).opacity(0.9),
            strokeColor: Color(.systemGray4).opacity(0.6)
        ) {
            VStack(alignment: .leading, spacing: 12) {
                HStack {
                    Text("Program Progress")
                        .font(.headline.weight(.semibold))
                        .foregroundColor(Color(.label))
                    Spacer()
                    Text(status.uppercased())
                        .font(.caption.weight(.semibold))
                        .foregroundColor(.appOrange)
                        .padding(.horizontal, 8)
                        .padding(.vertical, 4)
                        .background(Capsule().fill(Color.appOrangeLight))
                }

                Spacer(minLength: 0)

                HStack {
                    Spacer()
                    ZStack {
                        CompletionRing(progress: progress)
                            .frame(width: 140, height: 140)
                        VStack(spacing: 6) {
                            Text("\(progress)%")
                                .font(.title.weight(.bold))
                                .foregroundColor(Color(.label))
                            Text("\(elapsedDays)/\(totalDays) days")
                                .font(.subheadline.weight(.semibold))
                                .foregroundColor(Color(.secondaryLabel))
                        }
                    }
                    Spacer()
                }

                Spacer(minLength: 0)
            }
            .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topLeading)
        }
    }
}

struct CompletionRing: View {
    let progress: Int

    var body: some View {
        ZStack {
            Circle()
                .stroke(Color.black.opacity(0.5), lineWidth: 10)
            Circle()
                .trim(from: 0, to: CGFloat(max(min(Double(progress), 100), 0)) / 100.0)
                .stroke(
                    LinearGradient(
                        colors: [Color.appOrange, Color.appOrangeGradientEnd],
                        startPoint: .topLeading,
                        endPoint: .bottomTrailing
                    ),
                    style: StrokeStyle(lineWidth: 12, lineCap: .round)
                )
                .rotationEffect(.degrees(-90))
        }
    }
}

// MARK: - MTD metric cards

struct MTDParticipationCard: View {
    let active: Int
    let total: Int
    let pct: Double
    let change: Double

    var body: some View {
        CardShell(
            background: Color(.systemBackground).opacity(0.9),
            strokeColor: Color(.systemGray4).opacity(0.6)
        ) {
            VStack(alignment: .leading, spacing: 0) {
                Text("MTD Participation")
                    .font(.subheadline.weight(.semibold))
                    .foregroundColor(Color(.label))

                Spacer()

                VStack(alignment: .leading, spacing: 6) {
                    Text("\(Int(pct))%")
                        .font(.title2.weight(.bold))
                        .foregroundColor(Color(.label))

                    Text("\(active)/\(total) members active")
                        .font(.footnote.weight(.semibold))
                        .foregroundColor(Color(.secondaryLabel))
                }

                Spacer()

                VStack(alignment: .leading, spacing: 6) {
                    changeBadge
                    Text("vs prior MTD")
                        .font(.footnote)
                        .foregroundColor(Color(.secondaryLabel))
                }
            }
            .frame(maxWidth: .infinity, alignment: .leading)
        }
    }

    private var changeBadge: some View {
        let isUp = change >= 0
        let color: Color = isUp ? .green : .red
        let symbol = isUp ? "arrow.up" : "arrow.down"
        return HStack(spacing: 4) {
            Image(systemName: symbol)
            Text(String(format: "%.1f%%", abs(change)))
        }
        .font(.footnote.weight(.semibold))
        .padding(.horizontal, 10)
        .padding(.vertical, 6)
        .background(color.opacity(0.12))
        .foregroundColor(color)
        .cornerRadius(10)
    }
}

struct TotalWorkoutsCard: View {
    let total: Int
    let change: Double

    var body: some View {
        CardShell(
            background: Color(.systemBackground).opacity(0.9),
            strokeColor: Color(.systemGray4).opacity(0.6)
        ) {
            VStack(alignment: .leading, spacing: 0) {
                Text("Total workouts")
                    .font(.subheadline.weight(.semibold))
                    .foregroundColor(Color(.label))

                Spacer()

                Text("\(total)")
                    .font(.title2.weight(.bold))
                    .foregroundColor(Color(.label))

                Spacer()

                VStack(alignment: .leading, spacing: 6) {
                    changeBadge
                    Text("vs prior MTD")
                        .font(.footnote)
                        .foregroundColor(Color(.secondaryLabel))
                }
            }
            .frame(maxWidth: .infinity, alignment: .leading)
        }
    }

    private var changeBadge: some View {
        let isUp = change >= 0
        let color: Color = isUp ? .green : .red
        let symbol = isUp ? "arrow.up" : "arrow.down"
        return HStack(spacing: 4) {
            Image(systemName: symbol)
            Text(String(format: "%.1f%%", abs(change)))
        }
        .font(.footnote.weight(.semibold))
        .padding(.horizontal, 10)
        .padding(.vertical, 6)
        .background(color.opacity(0.12))
        .foregroundColor(color)
        .cornerRadius(10)
    }
}

struct TotalDurationCard: View {
    let hours: Double
    let change: Double

    var body: some View {
        CardShell(
            background: Color(.systemBackground).opacity(0.9),
            strokeColor: Color(.systemGray4).opacity(0.6)
        ) {
            VStack(alignment: .leading, spacing: 0) {
                Text("Total duration")
                    .font(.subheadline.weight(.semibold))
                    .foregroundColor(Color(.label))

                Spacer()

                Text("\(formattedHours) hrs")
                    .font(.title2.weight(.bold))
                    .foregroundColor(Color(.label))

                Spacer()

                VStack(alignment: .leading, spacing: 6) {
                    changeBadge
                    Text("vs prior MTD")
                        .font(.footnote)
                        .foregroundColor(Color(.secondaryLabel))
                }
            }
            .frame(maxWidth: .infinity, alignment: .leading)
        }
    }

    private var changeBadge: some View {
        let isUp = change >= 0
        let color: Color = isUp ? .green : .red
        let symbol = isUp ? "arrow.up" : "arrow.down"
        return HStack(spacing: 4) {
            Image(systemName: symbol)
            Text(String(format: "%.1f%%", abs(change)))
        }
        .font(.footnote.weight(.semibold))
        .padding(.horizontal, 10)
        .padding(.vertical, 6)
        .background(color.opacity(0.12))
        .foregroundColor(color)
        .cornerRadius(10)
    }

    private var formattedHours: String {
        let whole = Int(hours)
        if abs(hours - Double(whole)) < 0.05 {
            return "\(whole)"
        }
        return String(format: "%.1f", hours)
    }
}

struct AvgDurationCard: View {
    let minutes: Int
    let change: Double

    var body: some View {
        CardShell(
            background: Color(.systemBackground).opacity(0.9),
            strokeColor: Color(.systemGray4).opacity(0.6)
        ) {
            VStack(alignment: .leading, spacing: 0) {
                Text("Avg duration")
                    .font(.subheadline.weight(.semibold))
                    .foregroundColor(Color(.label))

                Spacer()

                Text("\(minutes) mins")
                    .font(.title2.weight(.bold))
                    .foregroundColor(Color(.label))

                Spacer()

                VStack(alignment: .leading, spacing: 6) {
                    changeBadge
                    Text("vs prior MTD")
                        .font(.footnote)
                        .foregroundColor(Color(.secondaryLabel))
                }
            }
            .frame(maxWidth: .infinity, alignment: .leading)
        }
    }

    private var changeBadge: some View {
        let isUp = change >= 0
        let color: Color = isUp ? .green : .red
        let symbol = isUp ? "arrow.up" : "arrow.down"
        return HStack(spacing: 4) {
            Image(systemName: symbol)
            Text(String(format: "%.1f%%", abs(change)))
        }
        .font(.footnote.weight(.semibold))
        .padding(.horizontal, 10)
        .padding(.vertical, 6)
        .background(color.opacity(0.12))
        .foregroundColor(color)
        .cornerRadius(10)
    }
}

// MARK: - Action cards

struct AddWorkoutCard: View {
    var body: some View {
        VStack(alignment: .leading, spacing: 10) {
            HStack {
                ZStack {
                    Circle()
                        .fill(Color.white.opacity(0.2))
                        .frame(width: 36, height: 36)
                    Image(systemName: "plus")
                        .font(.system(size: 16, weight: .bold))
                        .foregroundColor(.black)
                }
                Spacer()
                Image(systemName: "chevron.right")
                    .foregroundColor(Color(.black).opacity(0.6))
                    .font(.subheadline.weight(.bold))
            }

            Text("Add workouts")
                .font(.title3.weight(.bold))
                .foregroundColor(.black)

            Text("Log one or many sessions at once and keep progress up to date.")
                .font(.subheadline)
                .foregroundColor(.black.opacity(0.65))
                .multilineTextAlignment(.leading)
                .padding(.bottom, 4)

            Spacer(minLength: 0)

            HStack(spacing: 10) {
                Capsule()
                    .fill(Color.appOrange)
                    .frame(height: 38)
                    .overlay(
                        Label("Log sessions", systemImage: "bolt.fill")
                            .font(.subheadline.weight(.semibold))
                            .foregroundColor(.black)
                            .padding(.horizontal, 14)
                    )
            }
        }
        .padding()
        .background(
            LinearGradient(
                colors: [Color.appOrange, Color.appOrangeGradientEnd],
                startPoint: .topLeading,
                endPoint: .bottomTrailing
            )
        )
        .mask(RoundedRectangle(cornerRadius: 20, style: .continuous))
        .overlay(
            RoundedRectangle(cornerRadius: 20, style: .continuous)
                .stroke(Color.appOrange.opacity(0.3), lineWidth: 1)
        )
        .frame(height: 230, alignment: .topLeading)
    }
}

struct AddDailyHealthCard: View {
    var body: some View {
        VStack(alignment: .leading, spacing: 10) {
            HStack {
                ZStack {
                    Circle()
                        .fill(Color.white.opacity(0.2))
                        .frame(width: 36, height: 36)
                    Image(systemName: "bed.double.fill")
                        .font(.system(size: 16, weight: .bold))
                        .foregroundColor(.white)
                }
                Spacer()
                Image(systemName: "chevron.right")
                    .foregroundColor(Color.white.opacity(0.7))
                    .font(.subheadline.weight(.bold))
            }

            Text("Log daily health")
                .font(.title3.weight(.bold))
                .foregroundColor(.white)

            Text("Track sleep, diet quality, and steps for the day.")
                .font(.subheadline)
                .foregroundColor(Color.white.opacity(0.75))
                .multilineTextAlignment(.leading)
                .padding(.bottom, 4)

            Spacer(minLength: 0)

            HStack(spacing: 10) {
                Capsule()
                    .fill(Color.white.opacity(0.2))
                    .frame(height: 38)
                    .overlay(
                        Label("Log day", systemImage: "plus")
                            .font(.subheadline.weight(.semibold))
                            .foregroundColor(.white)
                            .padding(.horizontal, 14)
                    )
            }
        }
        .padding()
        .background(
            LinearGradient(
                colors: [Color.appBlue, Color.appBlueLight],
                startPoint: .topLeading,
                endPoint: .bottomTrailing
            )
        )
        .mask(RoundedRectangle(cornerRadius: 20, style: .continuous))
        .overlay(
            RoundedRectangle(cornerRadius: 20, style: .continuous)
                .stroke(Color.appBlue.opacity(0.25), lineWidth: 1)
        )
        .frame(height: 230, alignment: .topLeading)
    }
}

// MARK: - Card ordering model

enum SummaryCardType: String, CaseIterable, Hashable {
    case addWorkout
    case addDailyHealth
    case programProgress
    case mtdParticipation
    case totalWorkouts
    case totalDuration
    case avgDuration
    case activityTimeline
    case distributionByDay
    case workoutTypes

    var span: Int {
        switch self {
        case .addWorkout, .addDailyHealth, .programProgress, .activityTimeline, .distributionByDay, .workoutTypes:
            return 2
        default:
            return 1
        }
    }

    var requiresFullWidth: Bool {
        switch self {
        case .programProgress, .activityTimeline, .addWorkout, .addDailyHealth, .distributionByDay, .workoutTypes:
            return true
        default:
            return false
        }
    }

    static var defaultOrder: [SummaryCardType] = [
        .programProgress,
        .addWorkout,
        .addDailyHealth,
        .mtdParticipation,
        .totalWorkouts,
        .totalDuration,
        .avgDuration,
        .activityTimeline,
        .distributionByDay,
        .workoutTypes
    ]
}

struct CardDropDelegate: DropDelegate {
    let item: SummaryCardType
    @Binding var items: [SummaryCardType]
    @Binding var dragging: SummaryCardType?
    let onReorder: () -> Void

    func dropEntered(info: DropInfo) {
        guard let dragging,
              dragging != item,
              let from = items.firstIndex(of: dragging),
              let to = items.firstIndex(of: item) else { return }
        withAnimation(.spring(response: 0.25, dampingFraction: 0.85)) {
            items.move(fromOffsets: IndexSet(integer: from), toOffset: to > from ? to + 1 : to)
        }
    }

    func dropUpdated(info: DropInfo) -> DropProposal? {
        DropProposal(operation: .move)
    }

    func performDrop(info: DropInfo) -> Bool {
        dragging = nil
        onReorder()
        return true
    }
}
